terraform/09-video-hearing-core/modules/AppReg/main.tf

```diff
@@ -312,3 +312,12 @@ resource "azuread_app_role_assignment" "groups" {
 }
 
 data "azuread_client_config" "current" {}
+
+data "azuread_service_principal" "vh_user_api" {
+  application_id = azuread_application.app_reg["vh-user-api"].application_id
+}
+
+resource "azuread_directory_role_assignment" "vh_user_api_password_reset" {
```

Comment:
Could this be made dynamic like this please?
https://github.com/hmcts/vh-shared-infrastructure/blob/e36a34589afd324c0341c9be9209dca127fa59da/terraform/09-video-hearing-core/modules/AppReg/main.tf#L307